integration/monai/executors/client_algo_executor.py

```diff
@@ -0,0 +1,213 @@
+            elif task_name == self.submit_model_task:
+                return self.submit_model(shareable, fl_ctx)
+            elif task_name == self.validate_task:
+                return self.validate(shareable, fl_ctx, abort_signal)
+            else:
+                self.log_error(fl_ctx, f"Could not handle task: {task_name}")
+                return make_reply(ReturnCode.TASK_UNKNOWN)
+        except Exception as e:
+            # Task execution error, return EXECUTION_EXCEPTION Shareable
+            self.log_exception(fl_ctx, f"client_algo execute exception: {e}")
+            return make_reply(ReturnCode.EXECUTION_EXCEPTION)
+
+    def train(self, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -> Shareable:
+        self.log_debug(fl_ctx, f"train abort signal: {abort_signal.triggered}")
+
+        shareable.set_header(AppConstants.VALIDATE_TYPE, ValidateType.BEFORE_TRAIN_VALIDATE)
+
+        test_report = self.client_algo.evaluate(exchangeobj_from_shareable(shareable))
+        test_key_metric = test_report.metrics.get(self.key_metric)
+        self.log_info(
+            fl_ctx, f"{self.client_id} reported key metric {self.key_metric}: {test_key_metric}"
+        )  # only return key metric here
+        validate_result = DXO(
+            data_kind=DataKind.METRICS, data={MetaKey.INITIAL_METRICS: test_key_metric}
+        ).to_shareable()
+
+        self.client_algo.train(exchangeobj_from_shareable(shareable), extra={ExtraItems.ABORT: abort_signal})
+        local_weights_eo = weights_to_numpy(self.client_algo.get_weights())
+
+        # check returned weights dict
+        if local_weights_eo.weights is None:
+            self.log_error(fl_ctx, "Returned exchange object doesn't contain weights.")
+            return make_reply(ReturnCode.ERROR)
+
+        if local_weights_eo.weight_type == WeightType.WEIGHTS:
+            data_kind = DataKind.WEIGHTS
+        elif local_weights_eo.weight_type == WeightType.WEIGHT_DIFF:
+            data_kind = DataKind.WEIGHT_DIFF
+        else:
+            self.log_error(
+                fl_ctx,
+                f"Returned `WeightType` not supported. Expected {WeightType.WEIGHTS} or {WeightType.WEIGHT_DIFF} but got {local_weights_eo.get_weight_type()}",
+            )
+            return make_reply(ReturnCode.ERROR)
+
+        # get the number of executed steps
+        statistics = local_weights_eo.statistics
+        executed_steps = statistics.get(FlStatistics.NUM_EXECUTED_ITERATIONS)
+        if executed_steps:
+            meta = {MetaKey.NUM_STEPS_CURRENT_ROUND: executed_steps}
+        else:
+            meta = None
+
+        # TODO: support returned optimizer state (DXO collection?)
+        train_result = DXO(data_kind=data_kind, data=local_weights_eo.weights, meta=meta).to_shareable()
+
+        if not (train_result and isinstance(train_result, Shareable)):
+            return make_reply(ReturnCode.EMPTY_RESULT)
```

Comment:
when local_weights_eo.weights is None,
train_result is not None

Reply:
I'm following the same logic as in https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/app_common/executors/learner_executor.py.

Reply:
When the `local_weights_eo.weights` is `None`.
The function `DXO.to_shareable()" will return a non-empty Shareable.

Your code 
`train_result = DXO(data_kind=data_kind, data=local_weights_eo.weights, meta=meta).to_shareable()`
is diff VS learner executor: https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/app_common/executors/learner_executor.py#L95

In there that line of code might return None as result.
So the check make sense there.